### PR TITLE
Changed a switch statement to an if to reduce compiler warnings

### PR DESCRIPTION
**Pull request by anadon**
Just to reduce compiler warnings.

### CODE REVIEW

**Comment by swatanabe** [comment]
AMDG

Which warnings on which compilers?  The original code
looks correct and doesn't seem to be doing anything
even remotely dodgy.

In Christ,
Steven Watanabe

**Comment by anadon** [comment]
It triggers on `-Wswitch-enum`.  Tested on gcc 8.3.

**Comment by swatanabe** [comment]
AMDG

On 4/25/19 11:37 AM, jrmarsha wrote:
> It triggers on `-Wswitch-enum`.  Tested on gcc 8.3.
> 

Well don't use -Wswitch-enum, then.  I am strongly
against working around this be any method other
than suppression.

In Christ,
Steven Watanabe

**Comment by anadon** [comment]
I disagree, and here is why.  Let's make a pro-con list of this kind of trivial change, taking into account the perspective of a user in addition to Boost and the compiler.

+fewer compiler warnings
~no change in correctness
~no change in performance
~no change in understandability
+makes leveraging the compiler warning less burdensome

There aren't really any negatives.  There may be a perceived negative for changing something to change it, but that's not really a problem.  But if someone wants to use the `-Wswitch-enum` in their project, who are we to make that more burdensome or full of clutter?  Working with compiler warnings like this is a virtue for users.

**Comment by jeking3** [comment]
Reading through the documentation for this optional warning, I understand the intention is to help identify cases where one adds to an enumeration and then does not update every switch statement, leaving the defaults.  So one way to deal with this without rewriting it is to put all the enumerations into the same case group as default.  That causes no changes to the logic at all, and makes it more friendly to this type of defensive warning.  However unless the default build jobs run with `-Wswitch-enum` and `-Werror` (which can be very difficult to achieve) nobody would notice if it ever broke again.

**Comment by anadon** [comment]
I'd argue that those alternatives are likely to be larger changes which do not provide as clean of an outcome.  There is no need or utility in keeping the older code as it was, but there is some utility in the simple change to an if to catch a single special case.

**Comment by swatanabe** [comment]
AMDG

This warning is completely insane.  Treating it as
a hard rule essentially amounts to "never use default"
which is in no way reasonable.  Anyone one who uses
it should expect to see a lot of false positives and
have some way of dealing with them (i.e. -isystem)

#pragma GCC diagnostic push
#pragma GCC diagnostic ignored -Wswitch-enum
...
#pragma GCC diagnostic pop

(It can go at the top of the file along with
the msvc suppression)

In Christ,
Steven Watanabe

**Comment by anadon** [comment]
That's really dismissive, and the pragma alternative is more complex than just structuring to avoid the warning.  It isn't worth the effort to just ignore or disable a warning.  It isn't that bad.

**Comment by swatanabe** [comment]
AMDG

On 4/26/19 4:56 PM, jrmarsha wrote:
> That's really dismissive,

Yep.

For bonus points: -Wswitch-enum -Wswitch-default

> and the pragma alternative is more complex than just structuring to avoid the warning.

The pragma is better because it's completely separate
from the main code.

>  It isn't worth the effort to just ignore or disable a warning.  It isn't that bad.
> 

In Christ,
Steven Watanabe

**Comment by pdimov** [comment]
I can't reproduce this warning. On what code does it trigger?

**Comment by anadon** [comment]
I was fixing some depreciation warnings, and figured I'd turn up the warning levels just to check that everything was squeaky clean.  Back from when I was asking you about adding compiler flags, actually.  This popped up from graph tests from that, traced it back here, looked small so I changed it to what is in the diff, tested, then made the PR.

**Comment by pdimov** [comment]
OK, but what code reproduces the warning?

**Comment by anadon** [comment]
@pdimov Full details here: https://gist.github.com/anadon/bca4cb245ba452682cb669d5d51e2f78

The parts of interest are lines 41-58, 68-85, 188-203, 818-834, 1385-1401, 1956-1972, 1988-2003, corresponding to `all_planar_input_files_test.cpp`, `parallel_edges_loops_test.cpp`, `mas_test.cpp`, `stoer_wagner_test.cpp`, `successive_shortest_path_nonnegative_weights_test.cpp`, `cycle_canceling_test.cpp`, `find_flow_cost_bundled_properties_and_named_params_test.cpp`, `max_flow_algorithms_bundled_properties_and_named_params.cpp`.

File: include/boost/function/function_base.hpp [hunk 1]
```diff
@@ -507,16 +507,12 @@ namespace boost {
                functor_manager_operation_type op)
         {
           typedef typename get_function_tag<functor_type>::type tag_type;
-          switch (op) {
-          case get_functor_type_tag:
+          if(op == get_functor_type_tag){
```
Comment (pdimov) [inline include/boost/function/function_base.hpp]: Please observe the existing whitespace conventions in this file; `if (...) {`, `} else {`.

**Pull request merged**